application/tests/librarian/dataset_test.py

```
@@ -0,0 +1,123 @@
+    def test_ids_are_unique(self):
+        ids = [r["id"] for r in self.rows]
+        self.assertEqual(len(ids), len(set(ids)))
+
+
+class TestDatasetDeterminism(unittest.TestCase):
+    """The committed JSON must re-derive identically from the DB."""
+
+    def test_build_check_matches_committed_dataset(self):
+        if not os.path.exists(_DB):
+            self.skipTest("standards_cache.sqlite not present")
+        result = subprocess.run(
```

Ruff (0.15.17)

[error] `subprocess` call: check for execution of untrusted input

(S603)